### DISCUSSION TITLE
`next/mdx` is not tree-shaking unused imports

### DISCUSSION THREAD

**Discussion by georgecartridge** [vercel/next.js#65638] [Help]
### Summary

I've started using `next/mdx` to implement an MDX based blog. Whilst analysing my bundle sizes, I noticed that all the routes for the blog had the exact same bundle size despite importing different components. 

I have a single file in which I export several different components like so:

```tsx
// src/components/mdx.tsx

import A from '@/components/A';
import B from '@/components/B';
import C from '@/components/C';

export function ComponentOne() {
    return(
        <A />
    );
}

export function ComponentTwo({ children }: any) {
    return(
        <B>{children}</B>
    );
}

export function ComponentThree() {
    return(
        <C />
    );
}
```
I then have a `page.mdx` file that looks like this

```tsx
// src/app/demo/page.mdx

import { ComponentOne } from '@/components/mdx';

Here's some text

<ComponentOne />
```

Even though I am only importing `ComponentOne`, the rest of the components in the `src/components/mdx.tsx` seem to get bundled as well. 

EDIT: I tried to test the standard app pages by changing the MDX page to `page.tsx` with the exact same imports. It doesn't appear to tree-shake the unused modules here either. Is this by design?

I'm currently using Next 14.2.3

**Comment by RhythmAgg** [comment]
are you using ComponentTwo, ComponentThree in some other page/route?
